src/Microsoft.AspNetCore.Sockets/Transports/WebSocketsTransport.cs

```diff
@@ -75,7 +75,8 @@ public WebSocketsTransport(IChannelConnection<Message> application, ILoggerFacto
             {
                 if (receiving.IsCanceled || receiving.IsFaulted)
                 {
-                    // The receiver faulted or cancelled. This means the client is probably broken. Just propagate the exception and exit
+                    // The receiver faulted or cancelled. This means the socket is probably broken. Abort the socket and propagate the exception
+                    socket.Abort();
```

Comment:
Because we had a network error receiving from the socket. If we don't abort, Dispose will fail because of the (possible) active consumer. Although, after some changes I made in `ExecuteAsync`, we may be able to do this in a `finally` block in there rather than forcing the consumer to do it.